### PR TITLE
generate requirements lockfile in ci upon requirements change & update lockfile

### PR DESCRIPTION
**Pull request by cmyui**
# Describe your changes

## Checklist
- [x] I've manually tested my code
- [x] The changes pass pre-commit checks (`make lint`)
- [x] The changes follow coding style

### CODE REVIEW

**Comment by cmyui** [comment]
@NiceAesth nvm ez

File: .github/workflows/lock-requirements.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: '3.9'
+          cache: 'pip'
+      - name: Install pipenv
+        run:  python -m pip install --upgrade pipenv wheel
+      # TODO: add caching of Pipfile.lock
```
Comment (cmyui) [inline .github/workflows/lock-requirements.yaml:20]: not high priority, this already runs quick and should not be used frequently

**Review by tsunyoku — APPROVED** [review]

**Comment by NiceAesth** [comment]
wouldn't it make a bit more sense to do it the other way around? (e.g. have dependabot solely update pipfile and generate reqs from that)

changes look fine other than that

**Comment by cmyui** [comment]
> wouldn't it make a bit more sense to do it the other way around? (e.g. have dependabot solely update pipfile and generate reqs from that)

Yeah either way seems fine - didn't know what was possible/just went for a quick solution

**Pull request merged**